framework/filters/PageCache.php

```diff
@@ -155,7 +158,7 @@ public function init()
      */
     public function beforeAction($action)
     {
-        if (!$this->enabled) {
+        if (($this->enabled instanceof \Closure && !call_user_func($this->enabled, $this)) || !$this->enabled) {
```

Comment:
This looks incorrect. If closure will return `false `, it will be used again in `!$this->enabled` (which will always evaluate to `false`). Although it may work, it looks weird and relies on a bit of luck on type juggling.

Reply:
`!call_user_func($this->enabled, $this)` so, if the closure returns `false`, it will become `true` in the statement and will not trigger the evaluation of the second statement, thus never checking `!$this->enabled`.

Reply:
And what will happen if closure return `true`?

Reply:
Ah snap, you are right. Didn't think that one through.

`if($this->enabled instanceof \Closure) {
    $this->enabled = call_user_func($this->enabled, $this);
}
if (!$this->enabled) {`

This probably would be a cleaner solution.